File: gems/decomplex/examples/lua/state-mesh.lua
StateMeshExample = {}
function StateMeshExample:initialize() self.a = 1; self.b = 2 end
function StateMeshExample:writer() self.a = 3 end
function StateMeshExample:reader() return self.a + self.b end
function StateMeshExample:a_alias() return self.a end

File: gems/decomplex/examples/go/state-mesh.go
package main
type StateMeshExample struct { a int; b int }
func (self *StateMeshExample) initialize() { self.a = 1; self.b = 2 }
func (self *StateMeshExample) writer() { self.a = 3 }
func (self *StateMeshExample) reader() int { return self.a + self.b }
func (self *StateMeshExample) a_alias() int { return self.a }
